extensions/iceberg/src/main/java/io/deephaven/iceberg/util/IcebergTools.java

```diff
@@ -0,0 +1,91 @@
+//
+// Copyright (c) 2016-2024 Deephaven Data Labs and Patent Pending
+//
+package io.deephaven.iceberg.util;
+
+import io.deephaven.engine.table.ColumnDefinition;
+import io.deephaven.engine.table.TableDefinition;
+import org.apache.iceberg.PartitionField;
+import org.apache.iceberg.PartitionSpec;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.types.Type;
+import org.apache.iceberg.types.Types;
+
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.util.*;
+import java.util.stream.Collectors;
+
+/**
+ * Tools for managing and manipulating tables on disk in parquet format.
+ */
+public class IcebergTools {
+
+    @SuppressWarnings("unused")
+    public static IcebergCatalog loadCatalog(final String name, final IcebergInstructions instructions) {
+        return new IcebergCatalog(name, instructions);
+    }
+
+    static TableDefinition fromSchema(final Schema schema, PartitionSpec partitionSpec) {
+        final Set<String> partitionNames =
+                partitionSpec.fields().stream().map(PartitionField::name).collect(Collectors.toSet());
+
+        final List<ColumnDefinition<?>> columns = new ArrayList<>();
+
+        for (final Types.NestedField field : schema.columns()) {
+            final String name = field.name();
+            final Type type = field.type();
+            try {
+                final io.deephaven.qst.type.Type<?> qstType = convertPrimitiveType(type);
+                final ColumnDefinition<?> column;
+                if (partitionNames.contains(name)) {
+                    column = ColumnDefinition.of(name, qstType).withPartitioning();
+                } else {
+                    column = ColumnDefinition.of(name, qstType);
+                }
+                columns.add(column);
+            } catch (UnsupportedOperationException e) {
+                // TODO: Currently will silently skip the column. Would it be better to skip and warn the user or
+                // break and declare failure? We don't have a mechanism for skipping columns, do we need an overload
+                // with a supplied table definition?
```

Comment:
In the Parquet case, we throw a TableDataException in case of unsupported columns, you can check `ParquetColumnLocation` and `ParquetSchemaReader` classes and search for `Unsupported`. I think we can keep the similar behavior here.
But best to get opinion from Ryan on this.